src/Xamarin.Android.Build.Tasks/Tasks/LinkAssembliesNoShrink.cs

```diff
@@ -0,0 +1,98 @@
+using Java.Interop.Tools.Cecil;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using Mono.Cecil;
+using Mono.Cecil.Cil;
+using System;
+using System.IO;
+
+namespace Xamarin.Android.Tasks
+{
+	/// <summary>
+	/// This task is for Debug builds where LinkMode=None, LinkAssemblies is for Release builds
+	/// </summary>
+	public class LinkAssembliesNoShrink : Task
+	{
+		/// <summary>
+		/// These are used so we have the full list of SearchDirectories
+		/// </summary>
+		[Required]
+		public ITaskItem [] ResolvedAssemblies { get; set; }
+
+		[Required]
+		public ITaskItem [] SourceFiles { get; set; }
+
+		[Required]
+		public ITaskItem [] DestinationFiles { get; set; }
+
+		public override bool Execute ()
+		{
+			if (SourceFiles.Length != DestinationFiles.Length)
+				throw new ArgumentException ("source and destination count mismatch");
+
+			var readerParameters = new ReaderParameters {
+				ReadSymbols = true,
+			};
+			var writerParameters = new WriterParameters {
+				DeterministicMvid = true,
+			};
+
+			using (var resolver = new DirectoryAssemblyResolver (this.CreateTaskLogger (), loadDebugSymbols: true, loadReaderParameters: readerParameters)) {
+				// Add SearchDirectories with ResolvedAssemblies
+				foreach (var assembly in ResolvedAssemblies) {
+					var path = Path.GetFullPath (Path.GetDirectoryName (assembly.ItemSpec));
+					if (!resolver.SearchDirectories.Contains (path))
+						resolver.SearchDirectories.Add (path);
+				}
+
+				// Run the FixAbstractMethodsStep
+				var step = new FixAbstractMethodsStep (resolver, Log);
+				for (int i = 0; i < SourceFiles.Length; i++) {
+					var source = SourceFiles [i];
+					var destination = DestinationFiles [i];
+
+					// Only run the step on "MonoAndroid" assemblies
+					if (!MonoAndroidHelper.IsSharedRuntimeAssembly (source.ItemSpec) && MonoAndroidHelper.IsMonoAndroidAssembly (source)) {
```

Comment:
should we care about the order of the `&&` here. C# will have an early out option in `if` statements, so the quickest check should be first in the list. We should probably try to remember that when reviewing code from here on :)

My guess is the BinarySearch will be faster than the `assembly.GetMetadata` call, correct?